### PR TITLE
PDCL-10238 - encoding the cookie value for destinations 

### PR DESCRIPTION
**Pull request by ninaceban**
## Description

Closing this #978 PR and reopening the current PR with the changes. Removed the `reactor-cookie` dependency and added the `js-cookie` dependency and the related code to expose and use the `withConverter` API to override the cookie converter.
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Improvement (non-breaking change which does not add functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html) or I'm an Adobe employee.
- [ ] I have made any necessary test changes and all tests pass.
- [ ] I have run the Sandbox successfully.

### CODE REVIEW

**Review by carterworks — APPROVED** [review]

File: src/components/Audiences/index.js [hunk 1]
```diff
@@ -15,7 +15,15 @@ import injectProcessDestinations from "./injectProcessDestinations";
 import injectProcessResponse from "./injectProcessResponse";
 
 const createAudiences = ({ logger, fireReferrerHideableImage }) => {
-  const loggingCookieJar = createLoggingCookieJar({ logger, cookieJar });
+  const cookieJarWithEncoding = cookieJar.withConverter({
```
Comment (jfkhoury) [inline src/components/Audiences/index.js:18]: Can we please add a comment explaining why we are adding a converter?

Comment (ninaceban) [reply src/components/Audiences/index.js:18]: sure

**Review by carterworks — COMMENTED** [review]
I was looking through the changelog for `js-cookie` v3 (since `@adobe/reactor-cookie` is v2.2.1). There is one breaking change that I'm worried about:

> - Removed built-in JSON support, i.e. getJSON() and automatic stringifying in set(): use Cookies.set('foo', JSON.stringify({ ... })) and JSON.parse(Cookies.get('foo')) instead.

Is there anywhere that we store JSON in a cookie?

**Comment by jfkhoury** [comment]
> I was looking through [the changelog for `js-cookie` v3](https://github.com/js-cookie/js-cookie/releases/tag/v3.0.0) (since `@adobe/reactor-cookie` is v2.2.1). There is one breaking change that I'm worried about:
> 
> > * Removed built-in JSON support, i.e. getJSON() and automatic stringifying in set(): use Cookies.set('foo', JSON.stringify({ ... })) and JSON.parse(Cookies.get('foo')) instead.
> 
> Is there anywhere that we store JSON in a cookie?

I believe so. I suggest we use the same exact version we were using in the reactor package, and create a Jira ticket to upgrade to the latest safely. Thoughts?

**Review by jonsnyder — APPROVED** [review]
Looks good. Just add the comment Joe suggested.

Can you open a PR to remove this line in the extension code? https://github.com/adobe/reactor-extension-alloy/blob/a568b2e69460a6ce8dbd9b00eb09760b623cced5/rollup.config.js#L33

**Pull request merged**